packages/data-models/src/treebank_data_item.js

```diff
@@ -0,0 +1,69 @@
+export default class TreebankDataItem {
+  constructor (elem) {
+    this.version = 0
+    this.app = null
+    this.sourceUrl = null
+    this.reference = null
+
+    /*
+      Treebank data on a page must have an element with the following obligatory data attributes:
+        data-alpheios_tb_app - the only app currently supported is 'perseids-treebank-template'
+        data-alpheios_tb_app_version - a version of a data format (the latest version is 1);
+        data-alpheios_tb_app_url - a schema of a treebank template URL;
+        data-alpheios_tb_ref - a reference that will be used to load data into the iframe initially
+      Example:
+        data-alpheios_tb_app="perseids-treebank-template"
+        data-alpheios_tb_app_version="1"
+        data-alpheios_tb_app_url="https://alpheios-project.github.io/treebank-template/embed/DOC/SENTENCE?w=WORD"
+        data-alpheios_tb_ref="on-the-murder-of-eratosthenes-1-50#1-1"
+
+      HTML elements that are surrounding words must have a `data-alpheios_tb_ref` ref attribute.
+      It will tie a word to its position in a tree.
+      */
+    const tbSrcElem = elem.ownerDocument.querySelector('[data-alpheios_tb_app]')
+    if (tbSrcElem) {
+      this.app = tbSrcElem.dataset.alpheios_tb_app
+      if (this.app !== 'perseids-treebank-template') {
+        throw new Error('Unsupported treebank application. This version of Alpheios only supports the perseids-treebank-template viewer app.')
+      }
+      if (tbSrcElem.dataset.alpheios_tb_app_version) {
+        this.version = Number.parseInt(tbSrcElem.dataset.alpheios_tb_app_version, 10)
+        if (Number.isNaN(this.version)) { throw new Error(`Treebank version is incorrect in: ${tbSrcElem.outerHTML}`) }
+      }
+
+      if (!tbSrcElem.dataset.alpheios_tb_app_url) { throw new Error(`Missing treebank source URL in: ${tbSrcElem.outerHTML}`) }
+      this.sourceUrl = tbSrcElem.dataset.alpheios_tb_app_url
+
```

Comment:
as we discussed, when we eliminated support for the old treebank code, I wanted to use new attributes so that a page that was setup to work with the prior Alpheios release could be updated ahead of the Alpheios release, so that it could work with both the prior release of Alpheios and this one.

So for now I have kept the data-alpheios_tb_ref attribute, whose syntax has not changed and created these new attributes only understood by this version of the code

data-alpheios_tb_app
data-alpheios_tb_app_version
data-alpheios_tb_app_url

We can tweak these later if we need to.

Reply:
Great, thanks for letting me know!